shell/platform/windows/window_state.h

```diff
@@ -0,0 +1,73 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_WINDOWS_FLUTTER_WINDOW_STATE_H_
+#define FLUTTER_SHELL_PLATFORM_WINDOWS_FLUTTER_WINDOW_STATE_H_
+
+#include "flutter/shell/platform/common/cpp/client_wrapper/include/flutter/plugin_registrar.h"
+#include "flutter/shell/platform/common/cpp/incoming_message_dispatcher.h"
+#include "flutter/shell/platform/embedder/embedder.h"
+#include "flutter/shell/platform/windows/key_event_handler.h"
+#include "flutter/shell/platform/windows/keyboard_hook_handler.h"
+#include "flutter/shell/platform/windows/platform_handler.h"
+#include "flutter/shell/platform/windows/text_input_plugin.h"
+
+struct flutter::FlutterWindow;
+
+// Struct for storing state within an instance of the windows native (HWND or
+// CoreWindow) Window.
+struct FlutterDesktopWindowControllerState {
```

Comment:
I'm concerned by seeing these in a header. They are an unfortunate outcome of incremental growth of the GLFW implementation, and should be substantially refactored there; mirroring them in this new code is livable as a starting point, but making them public within all the code here is going to make it much harder to unwind this structure. If you are finding that you need elements of these things in other classes, please make a class-based design supporting those needs rather than further perpetuating the GLFW code's structure.

Reply:
Ignore for now. Since we'll be substantially altering the classes related to this, there will be clear opportunities to refactor and eliminate the structs; it doesn't need to block this PR.